api-lib/handlerHelpers.ts

```diff
@@ -0,0 +1,18 @@
+import { VercelRequest } from '@vercel/node';
+import { z } from 'zod';
+
+import {
+  InputSchema,
+  composeHasuraActionRequestBodyWithSession,
+  HasuraUserSessionVariables,
+} from '../src/lib/zod';
+
+export function getPropsWithUserSession<T extends z.ZodRawShape>(
```

Comment:
@topocount what do you think of this lil guy?

Reply:
@crabsinger this is an attempt at reducing action boilerplate. i first started with the idea of wrapping the entire handler function and passing these parsed props into it, but getting typing working was surprisingly difficult